grpcutil/dns_resolver.go

```diff
@@ -101,8 +101,11 @@ func parseTarget(target string) (host, port string, err error) {
 	return "", "", fmt.Errorf("invalid target address %v", target)
 }
 
-// Resolve creates a watcher that watches the name resolution of the target.
-func (r *dnsResolver) Resolve(target string) (Watcher, error) {
+// Resolve creates a watcher that watches the SRV/hostname record resolution of the target.
+//
+// First SRV record resolution is attempted for the specified service; if this fails, hostname record resolution is
+// attempted instead.
```

Comment:
```suggestion
// If service is not empty, watcher will first attempt to resolve SRV record.
// If that fails or service was empty, hostname record resolution is attempted instead.
```

Reply:
@pstibrany you probably want to say "If that fails, hostname record resolution is attempted instead." because your edited comment says that the watcher will only try SRV if service is non-empty.